### PR TITLE
[SDFAB-1092] Slice meter APIs on UP4

### PR DESCRIPTION
**Pull request by daniele-moro**
Currently, we support a single meter for mobile slices. This meter is shared between UL and DL.
We also removed the `no_tc` actions, so pfcp-agent must always specify a traffic class and we added the slice_tc meter to the logical pipeline.

Depends on:
- https://gerrit.onosproject.org/c/onos/+/25410
- https://github.com/stratum/fabric-tna/pull/503

### CODE REVIEW

File: p4src/main.p4 [hunk 1]
```diff
@@ -346,6 +349,23 @@ control PreQosPipe (inout parsed_headers_t    hdr,
         }
     }
 
+    action set_default_tc(tc_t tc) {
+        local_meta.tc = tc;
+    }
+
+    table default_tc {
```
Comment (daniele-moro) [inline p4src/main.p4:356]: remove default TC, and enforce pfcp to always provide TC in terminations actions

File: .env.stable [hunk 2]
```diff
@@ -2,8 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-PFCP_AGENT_IMAGE=omecproject/upf-epc-pfcpiface:master-f3a2c33
-ONOS_IMAGE=opennetworking/sdfabric-onos:master-2022-03-05
+PFCP_AGENT_IMAGE=dmoro92/upf-epc-pfcpiface:slice_meter
+ONOS_IMAGE=dmoro92/sdfabric-onos:slice_meter
```
Comment (daniele-moro) [inline .env.stable:6]: Need to be changed when we merge changes in ONOS, fabric-tna, and pfcp-agent.

File: scenarios/smf-setup.xml [hunk 3]
```diff
@@ -4,6 +4,10 @@
   -->
 <scenario name="smf-setup" description="Mock SMF setup">
     <group name="Smf-Setup">
+        <step name="Slice-Meter-Config"
+              exec="curl -X POST -H 'Content-Type:application/json' http://localhost:8080/v1/config/network-slices -d@./topo/sliceMeterConfig.json"/>
```
Comment (ccascone) [inline scenarios/smf-setup.xml:8]: to reduce verbosity consider creating a new command under bin to push the network slice config to pfcp-agent. Also, please make sure to respect existing case conventions (we don't use came case for files in the STC directory. I think we only use it for java class files)
```suggestion
              exec="pfcp-agent-push-slice-config topo/pfcp-slice-config.json"/>
```

File: scenarios/topo/sliceMeterConfig.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "sliceName" : "MOBILE",
```
Comment (ccascone) [inline scenarios/topo/sliceMeterConfig.json:2]: I know the name mobile slice is there for historical reasons, but lit's becoming too generic. `P4-UPF-1` would be a more accurate name considering that there will be more than one P4-UPF slice, and we plan to have slices to isolate BESS-UPF traffic. Eventually, we should replace all existing references of "mobile slice" in other parts of the codebase.

**Review by ccascone — APPROVED** [review]
LGTM. Feel free to merge when ready.

File: scenarios/topo/pfcp-slice-config.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "sliceName" : "MOBILE",
+  "sliceName" : "P4-UPF-1",
```
Comment (pierventre) [inline scenarios/topo/pfcp-slice-config.json:2]: do we need to verify if these params are actually set ?

**Review by pierventre — APPROVED** [review]

File: scenarios/topo/pfcp-slice-config.json (hunk 5 above)
Comment (daniele-moro) [reply scenarios/topo/pfcp-slice-config.json:2]: Added a verification of the meter band in the step where we verify the slice meter.

File: .env.stable [hunk 6]
```diff
@@ -2,8 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-PFCP_AGENT_IMAGE=omecproject/upf-epc-pfcpiface:master-f3a2c33
-ONOS_IMAGE=opennetworking/sdfabric-onos:master-2022-03-05
+PFCP_AGENT_IMAGE=omecproject/upf-epc-pfcpiface:master-762b242
+ONOS_IMAGE=dmoro92/sdfabric-onos:slice_meter
```
Comment (pierventre) [inline .env.stable:6]: is this ok ?

Comment (daniele-moro) [reply .env.stable:6]: Yes, unfortunately, we can't build a `sdfabric-onos` image right now with UP4 master (due to changes in ONOS APIs). For this reason, I had to build a custom `sdfabric-onos` image and publish that under my docker. I plan to merge it as is, trigger a sdfabric-onos build and then change this right after the image is published.

**Comment by daniele-moro** [comment]
retest this please

**Review by pierventre — APPROVED** [review]

**Comment by daniele-moro** [comment]
retest this please

**Comment by daniele-moro** [comment]
retest this please

**Comment by daniele-moro** [comment]
retest this please

**Comment by daniele-moro** [comment]
retest this please

**Pull request merged**